### DISCUSSION TITLE
Files compression - what options there are or will be?

### DISCUSSION THREAD

**Discussion by Szwendacz99** [longhorn/longhorn#9956] [Q&A]
Before I started using Longhorn, i enjoyed btrfs zstd compression very much, excluding large media files, the size was reduced ~50% overall. After migrating homelab to longhorn I suddenly need 4x more storage for those volumes without bigger media, since not only there is replication now, but also no compression.

Is or will be there any chance to compress file or filesystems on volumes, including encrypted ones?

**Comment by derekbit** [comment]
You can try `fsType: btrfs`.
https://longhorn.io/docs/1.7.2/references/storage-class-parameters/

**Reply by Szwendacz99** [reply]
> You can try `fsType: btrfs`. https://longhorn.io/docs/1.7.2/references/storage-class-parameters/

But then I would also need some option to specify mount options (to enable btrfs compression). Something like `nfsOptions` but for non-nfs mount. I don't see such option there.

**Reply by derekbit** [reply]
You can check the generic storageclass parameter `mountOptions`
- https://longhorn.io/docs/1.8.0/references/storage-class-parameters/#mount-options-field-mountoptions
- https://kubernetes.io/docs/concepts/storage/storage-classes/#mount-options

**Reply by Szwendacz99** [reply]
apparently I cannot use btrfs yet?

>Warning  FailedMount             68s (x10 over 5m18s)  kubelet                  MountVolume.MountDevice failed for volume "pvc-b56c40bd-9290-4a7a-a43e-09b9b07c999b" : rpc error: code = Internal desc = format of disk "/dev/mapper/pvc-b56c40bd-9290-4a7a-a43e-09b9b07c999b" failed: type:("btrfs") target:("/var/lib/kubelet/plugins/kubernetes.io/csi/driver.longhorn.io/9f74ef5b3d0ff218481c3256ce40f338337a4dc2a4cf8f9a47997a3315c8e43c/globalmount") options:("noatime,discard=async,compress=zstd:3,defaults") errcode:(executable file not found in $PATH) output:()
